util/topgen/merge.py

```diff
@@ -707,6 +715,31 @@ def amend_wkup(topcfg: OrderedDict):
         topcfg["inter_module"]["connect"]))
 
 
+# Handle reset requests from modules
+def amend_reset_request(topcfg: OrderedDict):
```

Comment:
this code is almost identical to amend `amend_wkup`. 
Merge the two together and just iterate over it twice.

Reply:
do you plan to do this in this PR?

Reply:
i think i might do it as a separate PR.